### PR TITLE
ARROW-13562: [R] Styler followups

### PR DESCRIPTION
**Pull request by nealrichardson**
Adds styling tasks to the Makefile (for 🦖  like me; I found that the styling-on-save from vscode was not reliable). Also makes codegen.R generate styled R code.

### CODE REVIEW

File: r/Makefile [hunk 1]
```diff
@@ -19,7 +19,13 @@ VERSION=$(shell grep ^Version DESCRIPTION | sed s/Version:\ //)
 ARROW_R_DEV="TRUE"
 ARROW_LARGE_MEMORY_TESTS=$(ARROW_R_DEV)
 
-doc:
+style:
+	R -s -e 'setwd(".."); if (requireNamespace("styler")) styler::style_file(setdiff(system("git diff --name-only | grep r/.*R$$", intern = TRUE), c("r/tests/testthat/latin1.R", "r/data-raw/codegen.R")))'
+
+style-all:
+	R -s -e 'styler::style_file(setdiff(dir(pattern = "R$$", recursive = TRUE), c("tests/testthat/latin1.R", "data-raw/codegen.R")))'
```
Comment (jonkeane) [inline r/Makefile:26]: We don't have to do this here, but given all where we are putting them, I wonder if we want to make (or investigate if this exists in styler already...) a file in r/arrow that specifies which files are no-style so we only have one place we need it

**Comment by lorenzwalthert** [comment]
> Adds styling tasks to the Makefile (for 🦖 like me; I found that the styling-on-save from vscode was not reliable). Also makes codegen.R generate styled R code.

Hi @nealrichardson, can you maybe tell us a bit more about problems with that so we could improve that?

**Comment by nealrichardson** [comment]
> > Adds styling tasks to the Makefile (for 🦖 like me; I found that the styling-on-save from vscode was not reliable). Also makes codegen.R generate styled R code.
> 
> Hi @nealrichardson, can you maybe tell us a bit more about problems with that so we could improve that?

@lorenzwalthert the first thing I found was that, while style-on-save would clearly do something (like remove extra whitespace), I was still getting lint warnings about files not having a trailing newline. I didn't get this if I just ran `styler` directly on the file. Later, it seemed like the style-on-save wasn't firing at all. I figured it was possible that either of those could just be user error, and I wanted a simple (to me) way to re-style all of the files anyway, so I added these `make` commands. If it would be helpful, I'm happy to write an issue somewhere.

**Comment by lorenzwalthert** [comment]
Thanks @nealrichardson. I referenced your comment in an open issue (https://github.com/REditorSupport/languageserver/issues/462), I hope this gets resolved. Regarding that formatting wasn't fired at all, I am unsure what the problem was. Just noted several issues in the above repo where the problem was the fact that styler is quite slow on unseen code (https://github.com/r-lib/styler/issues/558).

**Pull request closed without merging**